internal/components/fluentd/run_script.go

```diff
@@ -132,6 +132,46 @@ do
     fi
 done
 
+FILE="/var/lib/fluentd/pos/journal_pos.json"
+
+    echo "$FILE exists."
+
+if test -f "$FILE"; then
+    echo "$FILE exists."
+    ruby -v /etc/fluent/configs.d/user/cleanInValidJson.rb  2> /tmp/error-json-parser.log
+
+    if [ -s /tmp/error-json-parser.log ]; then
+      echo "its invalid json"
+      sudo rm $FILE
+    fi
+fi
+    echo "post checking json parser if working"
+
 exec fluentd $fluentdargs
 
 `
+
+const CleanInValidJson = `
+#!/usr/bin/env ruby
+
+require 'yajl'
+require 'json'
+
+
+FILE = "/var/lib/fluentd/pos/journal_pos.json"
```

Comment:
better to generate the file name from code. since this path is not a standard path. This must be same as path generated in fluent.conf.

Reply:
now passing FILE name as an argument to run.sh bash script. i can possibly grep pos files from fluent.conf and run over them a loop to check if they have a valid json file content or not.